omod/src/main/java/org/openmrs/module/aijar/web/controller/AppointmentCountController.java

```diff
@@ -0,0 +1,63 @@
+
+package org.openmrs.module.aijar.web.controller;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.aijar.AijarConstants;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+public class AppointmentCountController extends BaseRestController {
+        private final Log log = LogFactory.getLog(AppointmentCountController.class);
+
+        @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/" + AijarConstants.UGANDAEMR_MODULE_ID
+                        + "/appointmentcount", method = RequestMethod.GET)
+        @ResponseBody
+        public SimpleObject getNumberOfPatients(
+                        @RequestParam(required = true, value = "nextAppointmentDate") Date nextAppointmentDate,
+                        @RequestParam(value = "encounterTypeUuid", defaultValue = "") String encounterTypeUuid) {
+
+                AdministrationService administrationService = Context.getAdministrationService();
```

Comment:
@reagan-meant I asked you to autowire this service

Reply:
@ssmusoke Autowiring causes [this error](https://pastebin.com/5vj2HZuf) and also calling context from outside a method would cause [this error](https://talk.openmrs.org/t/initialising-a-dataset/27925/34) That is why I did it inside a method.....Is there a way to prevent that error?